Tutorial: How to build a Vue cookie consent banner (#5128)
PostHog/posthog.com: contents/tutorials/vue-cookie-banner.md

---
title: Building a Vue cookie consent banner
date: 2023-01-20
author: ["ian-vanagas"]
showTitle: true
sidebar: Docs
featuredImage: ../images/tutorials/banners/posthog-for-vuejs.png
topics: ['configuration']
---

With internet privacy regulations, like GDPR, coming into effect, managing cookies is becoming increasingly important. Cookies are pieces of information apps set in users’ browsers to help them store information and identity. It’s possible to use [PostHog without cookies](/tutorials/cookieless-tracking), but it’s simpler to use them.

> To ensure you are compliant with regulations like GDPR, your app must receive consent to use cookies. The cookie consent banner is a way to do this, and this tutorial shows you how to build one in Vue, a popular JavaScript framework.

Reviewer: ```suggestion
To ensure you are compliant with regulations such as GDPR, your app must receive consent to use cookies. One way to do this is with a cookie consent banner, and this tutorial shows you how to build one in Vue, a popular JavaScript framework.
```

## Step 1: Setting up our Vue app

> First, if you haven’t used Vue before, install the command line interface (CLI). You also need a package manager like npm, which you can get by installing [Node](https://nodejs.org/en/).

Reviewer: ```suggestion
First, if you haven’t used Vue before, install the command line interface (CLI). You also need a package manager such as npm, which you can get by installing [Node](https://nodejs.org/en/).
```

```bash
npm install -g @vue/cli
```

> After installing the CLI, you can create the Vue project. We are naming ours "tutorial." When the CLI prompts you with options, choose `vue3` and `npm` (or yarn, it doesn’t matter).

Reviewer: ```suggestion
After installing the CLI, you can create the Vue project. We are naming ours "tutorial." When the CLI prompts you with options, choose `vue3` and `npm` (or `yarn`, it doesn’t matter).
```

npm i posthog-js
cd src
mkdir plugins
touch posthog.js
```

In `posthog.js`, add the code to set up and initialize PostHog as a plugin.

```js
// src/plugins/posthog.js
import posthog from "posthog-js";

export default {
  install(app, options) {
    app.config.globalProperties.$posthog = posthog.init(
      "<ph_project_api_key>",
      {
        api_host: "<ph_instance_address>",
      }
    );
  },
};
```

In `main.js`, import the PostHog plugin and connect it to our app.

```js
// main.js
import { createApp } from 'vue'
import App from './App.vue'
import posthogPlugin from "./plugins/posthog"; //import the plugin. 

const app = createApp(App);

app.use(posthogPlugin); //install the plugin
app.mount('#app')
```

Once you run `npm run serve` again and head into the cookies for your site, you should see a PostHog cookie added.

![Cookie](../images/tutorials/vue-cookie-banner/cookie.png)

Our app did this without the consent of the user, and now, it’s time to build the cookie consent banner to fix this.

## Step 3: Creating our cookie banner component

First, create a `CookieBanner.vue` component in our `components` folder.

```bash
cd components
touch CookieBanner.vue
```

> For our cookie banner, create a basic component with some details about cookies, buttons to accept and decline, and some basic styling.

Reviewer: May be fun to reference the file we use in our repo here, so users can see what the messaging should be?

Author: Yup, added 😄
